cincinnati/src/plugins/internal/graph_builder/release_scrape_dockerv2/registry/mod.rs

```diff
@@ -208,9 +208,10 @@ pub async fn fetch_releases(
     cache: cache::Cache,
     manifestref_key: &str,
     concurrency: usize,
+    force_auth: bool,
 ) -> Result<Vec<cincinnati::plugins::internal::graph_builder::release::Release>, Error> {
     let registry_client = {
-        let authenticate = username.is_some() && password.is_some();
+        let authenticate = force_auth || (username.is_some() && password.is_some());
```

Comment:
Which username and password are sent when `force_auth` is true?

Reply:
The ones which are set, which could be none.

Reply:
In that case why can't `None` be sent right away, trap 401 and try without username/password?

Reply:
I'll try to fill in your question:
In the case of username and password being `None`, we try to fetch the releases without authentication first, and if it tails with a 401 error code we try to get a token anonymously and then retry the release fetching.
I think that can be done. I have just kept it simpler and left the decision for anonymous authentication to the congiuration.

Reply:
To drive this conversation towards resolve, @vrutkovs are you asking for a change in the above described manner or were you merely curious?